inference-engine/src/mkldnn_plugin/nodes/gather_nd.cpp

```diff
@@ -0,0 +1,230 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "base.hpp"
+
+#include <string>
+#include <vector>
+#include "ie_parallel.hpp"
+#include "common/cpu_memcpy.h"
+
+namespace InferenceEngine {
+namespace Extensions {
+namespace Cpu {
+
+class GatherNDImpl: public ExtLayerBase {
+public:
+    explicit GatherNDImpl(const CNNLayer* layer) {
+        _errorPrefix = std::string("Layer GatherND with name '") + layer->name + "'";
+
+        if (layer->insData.size() != 2 || layer->outData.size() != 1)
+            THROW_IE_EXCEPTION << _errorPrefix << " has invalid number of input/output edges.";
+
+        auto data = layer->insData[_dataIndex].lock();
+        auto indices = layer->insData[_indicesIndex].lock();
+        if (!data || !indices)
+            THROW_IE_EXCEPTION << _errorPrefix << " has nullable inputs.";
+        Precision dataPrecision = data->getTensorDesc().getPrecision();
+        if (dataPrecision.size() != sizeof(PrecisionTrait<Precision::I32>::value_type) &&
```

Comment:
if you add I64 to test looks like should add check for I64 here

Reply:
Here could not be I64. It is converted to I32 before.